src/utilities.jl

```diff
@@ -198,3 +198,184 @@ eof(c::Char) = c === EOF_CHAR
+    0x00002944 <= c <= 0x00002970 ||
+    0x000029b7 <= c <= 0x000029b8 ||
+    c == 0x000029bc ||
+    0x000029be <= c <= 0x000029c1 ||
+    c == 0x000029e1 ||
+    0x000029e3 <= c <= 0x000029e5 ||
+    c == 0x000029f4 ||
+    0x000029f6 <= c <= 0x000029f7 ||
+    0x000029fa <= c <= 0x000029fb ||
+    0x00002a07 <= c <= 0x00002a08 ||
+    c == 0x00002a1d ||
+    0x00002a22 <= c <= 0x00002a2e ||
+    0x00002a30 <= c <= 0x00002a3d ||
+    0x00002a40 <= c <= 0x00002a45 ||
+    0x00002a4a <= c <= 0x00002a58 ||
+    0x00002a5a <= c <= 0x00002a63 ||
+    0x00002a66 <= c <= 0x00002a67 ||
+    0x00002a6a <= c <= 0x00002ad9 ||
+    c == 0x00002adb ||
+    0x00002af7 <= c <= 0x00002afa ||
+    0x00002b30 <= c <= 0x00002b44 ||
+    0x00002b47 <= c <= 0x00002b4c ||
+    0x0000ffe9 <= c <= 0x0000ffec
+end
+
+
+@inline function isopsuffix(c1::Char)
+    c1 == EOF_CHAR && return false
+    c = UInt32(c1)
+    0x000000b2 <= c <= 0x000000b3 ||
+    c == 0x000000b9 ||
+    c == 0x000002b0 ||
+    0x000002b2 <= c <= 0x000002b3 ||
+    0x000002b7 <= c <= 0x000002b8 ||
+    0x000002e1 <= c <= 0x000002e3 ||
+    c == 0x00001d2c ||
+    c == 0x00001d2e ||
+    0x00001d30 <= c <= 0x00001d31 ||
+    0x00001d33 <= c <= 0x00001d3a ||
+    c == 0x00001d3c ||
+    0x00001d3e <= c <= 0x00001d43 ||
+    0x00001d47 <= c <= 0x00001d49 ||
+    c == 0x00001d4d ||
+    0x00001d4f <= c <= 0x00001d50 ||
+    c == 0x00001d52 ||
+    0x00001d56 <= c <= 0x00001d58 ||
+    c == 0x00001d5b ||
+    0x00001d5d <= c <= 0x00001d6a ||
+    c == 0x00001d9c ||
+    c == 0x00001da0 ||
+    0x00001da5 <= c <= 0x00001da6 ||
+    c == 0x00001dab ||
+    c == 0x00001db0 ||
+    c == 0x00001db8 ||
+    c == 0x00001dbb ||
+    c == 0x00001dbf ||
+    c == 0x00002009 ||
+    0x00002032 <= c <= 0x00002037 ||
+    c == 0x00002057 ||
+    0x00002070 <= c <= 0x00002071 ||
+    0x00002074 <= c <= 0x0000208e ||
+    0x00002090 <= c <= 0x00002093 ||
+    0x00002095 <= c <= 0x0000209c ||
+    0x00002c7c <= c <= 0x00002c7d
+end
+
+
+function optakessuffix(k)
+    (Tokens.begin_ops < k < Tokens.end_ops) && 
+    !(k == Tokens.DDDOT ||
+    Tokens.EQ <= k <= k == Tokens.XOR_EQ ||
+    k == Tokens.CONDITIONAL ||
+    k == Tokens.RIGHT_ARROW ||
+    k == Tokens.LAZY_OR ||
+    k == Tokens.LAZY_AND ||
+    k == Tokens.ISSUBTYPE ||
+    k == Tokens.ISSUPERTYPE ||
+    k == Tokens.IN ||
+    k == Tokens.ISA ||
+    k == Tokens.COLON_EQUALS ||
+    k == Tokens.DOUBLE_COLON_EQUAL ||
+    k == Tokens.COLON ||
+    k == Tokens.DDOT ||
+    k == Tokens.EX_OR ||
+    k == Tokens.DECLARATION ||
+    k == Tokens.WHERE ||
+    k == Tokens.DOT ||
+    k == Tokens.NOT ||
+    k == Tokens.TRANSPOSE ||
+    k == Tokens.ANON_FUNC ||
+    Tokens.NOT_SIGN <= k <= Tokens.QUAD_ROOT
+    ) 
+end
+
+# suffix operators
+# "₀₁₂₃₄₅₆₇₈₉₊₋₌₍₎²³¹ʰʲʳʷʸˡˢˣᴬᴮᴰᴱᴳᴴᴵᴶᴷᴸᴹᴺᴼᴾᴿᵀᵁᵂᵃᵇᵈᵉᵍᵏᵐᵒᵖᵗᵘᵛᵝᵞᵟᵠᵡᵢᵣᵤᵥᵦᵧᵨᵩᵪᶜᶠᶥᶦᶫᶰᶸᶻᶿ ⁰ⁱ⁴⁵⁶⁷⁸⁹⁺⁻⁼⁽⁾ⁿₐₑₒₓₕₖₗₘₙₚₛₜⱼⱽ′″‴‵‶‷⁗"
+
+
+# Used to generate dotop1 and isopsuffix
+function _genchartest(x::Vector{UInt32}, fname)
```

Comment:
Oh, they are generated from here? What is the reason to write the function out manually?

Reply:
The input vectors are created by filtering the operator list according to how they parse ( I've forgotten to include the code), which we don't want to do on startup

Reply:
I thinks it's slightly more transparent this way and some of it was done by hand

Reply:
I also don't like having both the manual and the generated version in here. We should either have one or the other.

Reply:
fair enough, stored [here](https://gist.github.com/ZacLN/092430d37c5b31479b5c7bbb1d93bf82) for ref